### ISSUE TITLE
Unable to open project directory while it's removed

### ISSUE THREAD

**Issue by noe132** [atom/atom#16645]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: https://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Every time when open atom show error message that "Unable to open project directory", while a project folder was add to the sidebar before, but now is removed while atom wasn't opened, and it's not able to remove that project cause it's not showed in sidebar.
![tim 20180130002122](https://user-images.githubusercontent.com/2271900/35521210-2421b544-0554-11e8-854a-6a8c8e13c922.png)

### Steps to Reproduce

1. Open Atom
2. Create a folder on disk somewhere
3. File -> Add Project Folder, Choose that folder we created
4. Close Atom
5. Delete that folder
6. Open Atom
7. it goes the message

**Expected behavior:** maybe notify the user to remove the project or not?

**Actual behavior:** only goes the error message. If I want to remove that, I have to recreate that folder and open atom to remove it from the sidebar

**Reproduces how often:** everytime

### Versions

Atom    : 1.23.3
Electron: 1.6.15
Chrome  : 56.0.2924.87
Node    : 7.4.0

apm  1.18.12
npm  3.10.10
node 6.9.5 x64
atom 1.23.3
python 2.7.13
git 2.12.2.windows.2
visual studio 2015

windows 10 x64 1709

**Comment by rsese** [comment]
Thanks for the report!  I can reproduce on macOS 10.12.6 1.23.3, I'll tentatively tag as a bug for now and ask the team to take a look.

**Comment by pabloab** [comment]
**Workaround**: remove nonexistent path from JSON array on `~/.atom/storage/application.json`, on `initialPaths`.
Also discussed [here](https://discuss.atom.io/t/unable-to-open-project-directory/49477/).

**Comment by MatthewHerbst** [comment]
I am having this same issue. It started after I did a `git checkout` to a different branch of the codebase I work on.

**Comment by draeath** [comment]
I've got the same on Linux, with Atom 1.24.0.

Currently, it just doesn't add the item to the list (but whines in an error notification on start.

The project sidebar ideally should list all configured projects, and in the case where the path is no longer valid this should be made visible. This would allow the user to see what was missing once the notification times out, and take appropriate action (such as removing the project)

Currently, one thing _is doing right_ is **not** automatically removing the project. Please do not implement this as a quick fix, as it masks an underlying issue (the directory being missing). Perhaps it was renamed, or mistakenly removed. Perhaps it's a in a (currently unavailable) filesystem mount. Any of these cases, one would not want it automatically removed.

**Comment by rub1sco** [comment]
Had the same issue as well. I was able to clear the message by adding the folder back to the directory and deleting it from atom, then deleting it from the folder. It'd be great for a fix though!

**Comment by MisterSpicy** [comment]
Same issue on a new Atom install - looking at the ~/.atom/storage/application.json, I have no paths listed:
```
[{"initialPaths":[]}]
```

**Comment by tandriamil** [comment]
Same for me here, it is quite annoying seeing this error message everyday for a folder that I deleted a week ago...

Maybe a warning would be better than an error message, and let the project directory into the tree view but with a visual information about its missing state? Which will allow each user to decide what to do, deleting it or trying to restore it.

**Comment by FranekW** [comment]
I have got a similar error (without the code):
"Unable to open project directory
Project directory Z:\aaa\xxx\yyy is no longer on disk."

On WIndows 10, I have followed those steps:
1. Create a folder.
2. In Atom, FIle -> Open Folder... -> and choose the folder just created
3. Atom should run and open another window with a project pointing to the folder (two Atom windows should exist in a taskbar)
4. Close all Atom windows but make sure the one that's pointing to the folder is the last
5. Delete the folder
6. If you open Atom, it will complain that project no longer exists.

Besides that, application.json should contain "initialPath" pointing at the folder created before.
